x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -222,8 +272,9 @@ export class MBMapContainer extends React.Component {
   };
 
   render() {
-    // do not debounce syncing zoom and center
+    // do not debounce syncing of map-state
```

Comment:
It has always bugged me that these side-effect methods get called from render. Can you move these to `componentDidUpdate`?